### PR TITLE
feat(hycu): add listing page

### PR DESCRIPTION
**Pull request by tibs245**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/hycu`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Feat #MANAGER-14497
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

### CODE REVIEW

File: packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json [hunk 1]
```diff
@@ -1,4 +1,16 @@
 {
-  "title": "Listing page",
-  "listing_resultats": "résultats"
+  "title": "HYCU",
+  "listing_resultats": "résultats",
+  "hycu_status_activated": "Active",
+  "hycu_status_toActivate": "A activer",
```
Comment (pauldkn) [inline packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json:5]: **typo:** À ?

File: packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+  "hycu_status_pending": "En cours d'activation",
+  "hycu_status_error": "Error d'activation",
```
Comment (pauldkn) [inline packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json:7]: **typo:** Erreur

File: packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json [hunk 3]
```diff
@@ -1,4 +1,16 @@
...
+  "hycu-cloud-vm-pack-unknown": "Pack inconnu",
+  "hycu_name": "Nom",
+  "hycu_controller_id": "Controller ID",
+  "hycu_status": "Status",
```
Comment (pauldkn) [inline packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json:11]: **typo:** Statut ?

File: packages/manager/apps/hycu/package.json
Comment (darsene) [inline packages/manager/apps/hycu/package.json:1]: file yarn.lock should be modified as well normally

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx [hunk 4]
```diff
@@ -1,121 +1,216 @@
...
+  const { data: serviceDetails, isLoading } = useServiceDetails({
+    resourceName: hycuDetail.serviceName,
+  });
+
+  const productName = usePackTypeLabel(
+    serviceDetails?.data.resource.product.name,
+  );
+
+  return (
+    <DataGridTextCell>{isLoading ? 'Loading' : productName}</DataGridTextCell>
```
Comment (pauldkn) [inline packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:92]: Hardcoded "Loading" text : should be using translations ?

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx [hunk 5]
```diff
@@ -1,121 +1,216 @@
...
+  const { data: serviceDetails, isLoading } = useServiceDetails({
+    resourceName: hycuDetail.serviceName,
+  });
+  const date = new Date(
+    serviceDetails?.data.billing.lifecycle.current.creationDate,
+  );
+  const formattedDate = useFormattedDate({ date, options: dateFormat });
+
+  return (
+    <DataGridTextCell>{isLoading ? 'Loading' : formattedDate}</DataGridTextCell>
```
Comment (pauldkn) [inline packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:106]: Same for hardcoded "loading"

File: packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json (hunk 3 above)
Comment (darsene) [inline packages/manager/apps/hycu/public/translations/listing/Messages_fr_FR.json:11]:
```suggestion
  "hycu_status": "Statut",
```

File: packages/manager/apps/hycu/src/hooks/useFormattedDate.ts
Comment (darsene) [inline packages/manager/apps/hycu/src/hooks/useFormattedDate.ts:1]: This hook already exists in manager-components, right ?

File: packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx [hunk 6]
```diff
@@ -1,121 +1,216 @@
...
+                size={ODS_BUTTON_SIZE.sm}
+                inline
+              >
+                {t('hycu_order')}
+              </OsdsButton>
+            </div>
+            {columns && flattenData && (
+              <Datagrid
+                columns={columns}
+                items={flattenData || []}
```
Comment (darsene) [inline packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx:203]: `flattenData` can't be null here
```suggestion
                items={flattenData}
```

File: packages/manager/apps/hycu/src/hooks/services/usePackTypeLabel.ts
Comment (darsene) [inline packages/manager/apps/hycu/src/hooks/services/usePackTypeLabel.ts:1]: You could add unit test for that hook

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/apps/hycu/package.json
Comment (tibs245) [reply packages/manager/apps/hycu/package.json:1]: I think I already pushed change on precedent PR by accident with multiple checkout. I have no change on yarn install

File: packages/manager/apps/hycu/src/hooks/services/usePackTypeLabel.ts
Comment (tibs245) [reply packages/manager/apps/hycu/src/hooks/services/usePackTypeLabel.ts:1]: I duplicated this hook by accident. Deleted :)

**Review by pauldkn — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**